use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/service/TimeTableDemoResource.java

```diff
@@ -0,0 +1,217 @@
+package org.acme.schooltimetabling.service;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.acme.schooltimetabling.domain.Lesson;
+import org.acme.schooltimetabling.domain.Room;
+import org.acme.schooltimetabling.domain.TimeTable;
+import org.acme.schooltimetabling.domain.Timeslot;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+
+@Tag(name = "Demo data sets", description = "Timefold-provided demo school timetable data sets.")
+@Path("demo/datasets")
+public class TimeTableDemoResource {
+
+    public enum DemoDataSet {
+        SMALL,
+        LARGE
+    }
+
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "List of demo data sets represented as IDs.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = DemoDataSet.class, type = SchemaType.ARRAY))) })
+    @Operation(summary = "List demo data sets.")
+    @GET
+    public DemoDataSet[] list() {
+        return DemoDataSet.values();
+    }
+
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Unsolved demo timetable.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = TimeTable.class))),
+            @APIResponse(responseCode = "404", description = "Demo data set does not exist.",
+                    content = @Content(mediaType = MediaType.TEXT_PLAIN))})
+    @Operation(summary = "Find an unsolved demo timetable by ID.")
+    @GET
+    @Path("/{id}")
+    public Response generate(@Parameter(description = "Unique identifier of the demo data set.",
+            required = true) @PathParam("id") String dataSetId) {
```

Comment:
Nitpick: I prefer to keep the java parameter name and the url parameter the same name if they are the same
```suggestion
    @Path("/{dataSetId}")
    public Response generate(@Parameter(description = "Unique identifier of the demo data set.",
            required = true) @PathParam("dataSetId") String dataSetId) {
```

Reply:
You can also probably accept DemoDataSet as the type for dataSetId instead of String directly.

Reply:
Not without adding some resolver class that takes care of a non-existing enumeration item. Without it, `curl` returns nothing without saying what went wrong.